crates/sui-sdk/src/lib.rs

```diff
@@ -555,3 +558,101 @@ impl DataReader for ReadApi {
         Ok(self.get_reference_gas_price().await?)
     }
 }
+
+fn add_port_to_url(url: &str) -> Result<String, Error> {
+    let parsed_uri = reqwest::Url::parse(url)
+        .map_err(|_| Error::InvalidURI(format!("{}. Please provide an absolute URI", url)))?;
+
+    // if the URI is name.domain:port_number, then just return an error as the underlying
+    // http connection requires a scheme
+    if parsed_uri.cannot_be_a_base() {
+        return Err(Error::InvalidURI(format!(
+            "{}. URL scheme not supported, expects 'http', 'https' or 'wss'.",
+            url
+        )));
+    }
+
+    let mut port = 80;
+    if (parsed_uri.scheme() == "https" || parsed_uri.scheme() == "wss")
+        && parsed_uri.port().is_none()
+    {
+        port = 443;
+    }
```

Comment:
it is entirely possible that you looked into this and it doesn't work, but I came across [socket_addrs](https://docs.rs/reqwest/latest/reqwest/struct.Url.html#method.socket_addrs) and wonder if this is a better function for this logic?

Reply:
Interesting -- it could be that if you set the "known default" port, it just gets ignored? In which case, you could go the other way:

```rust
fn add_port_to_url(url: &str) -> Result<String, Error> {
    let mut parsed_url = reqwest::Url::parse(url)?;
    Ok(if let Some(port) = parsed_url.port_or_known_default() {
        format!("{parsed_url}:{port}")
    } else {
        format!("{parsed_url}")
    })
}
```

Reply:
This works well if the URL is `http://domain.com` but not for `http://domain.com/v1/faucet`

Reply:
>    #[error(transparent)]
>    InvalidUrl(#[from] reqwest::url::ParseError),

Unfortunately, this does not seem to work because reqwest does not re-export `url` crate, and `ParseError` comes from it.

Reply:
😢

Reply:
I'm sorry I need relax I came after and look on problems